### ISSUE TITLE
[LintDiff] Pin version of @microsoft.azure/openapi-validator-rulesets

### ISSUE THREAD

**Issue by mikeharder** [Azure/azure-sdk-tools#7613]
LintDiff pins the version of `@microsoft.azure/openapi-validator`, but it does not pin the version of `@microsoft.azure/openapi-validator-rulesets`.  `@microsoft.azure/openapi-validator` does not pin either.  The end result is LintDiff is using `@microsoft.azure/openapi-validator-rulesets@latest`, which can break any time a new version is published.

Instead, LintDiff should somehow pin the version of `@microsoft.azure/openapi-validator-rulesets`, so it is only updated via a PR merged to LintDiff.  LintDiff could pin the version directly, or `@microsoft.azure/openapi-validator` could pin instead of float.

**Comment by weshaggard** [comment]
I thought we weren't pinning by design because we wanted to make release ruleset updates by just release a new package. Of course if we do use that strategy then we need to be extra careful about releasing this ruleset package.

**Comment by konrad-jamrozik** [comment]
We are having this conversation because of this:
- https://github.com/Azure/azure-openapi-validator/issues/653

However, the `stagingOnly` system is not really well suited for this scenario. It is designed to update one rule art a time, but the problem was caused by schema change to all rulesets:
- https://github.com/Azure/azure-openapi-validator/pull/648

**Comment by mikeharder** [comment]
I think we can change how these packages are versioned and consumed, that will satisfy all our requirements but also follow the best practices of the JS ecosystem.

**Packages** should use proper semver for their own version (breaking changes increment major, additive changes increment minor, bug fixes increment patch).  Packages should typically depend on other packages using `^`, which floats to the latest compatible version.  There may be exceptions if you own both packages, where you choose to pin to guarantee package versions are aligned (which may be appropriate for the `openapi-validator-rulesets` package family).

**Applications** (like `Swagger LintDiff`) should have a `package.json` expressing their dependencies (pinned or floated), and a `package-lock.json` that pins the version of all direct and transitive dependencies.  Thus, application dependencies should never change at runtime.  Only via a PR that updates `package-lock.json`.

I understand `Swagger LintDiff` does not currently work this way, but we should start moving in this direction.

**Comment by konrad-jamrozik** [comment]
@mikeharder makes sense to me. Currently the way it works, on a high level, is that as soon a PR is merged, CI triggers and publishes new `beta` package without any approvals or what not. Then there is a separate CI check, "Swagger LintDiff (Staging)" (forgot exact title), that runs based-off the beta packages. Unfortunately, the source code for it was done by copy-paste-adapt, so the code behind this staging CI check is almost identical to the prod LintDiff, but naturally has a different AutoRest CLI invocation that relies on latest beta package.

**Comment by konrad-jamrozik** [comment]
I am going to make some improvements to the CI process. Tracking it here:
- https://github.com/Azure/azure-sdk-tools/issues/7619
